include/cblox/core/tsdf_submap_collection.h

```diff
@@ -28,115 +28,92 @@ class TsdfSubmapCollection {
 
   // Gets a vector of the linked IDs
-  void getIDs(std::vector<SubmapID> *submap_ids) const;
-  bool isBaseFrame(const SubmapID &submap_id) const;
+  std::vector<SubmapID> getIDs() const;
+  bool exists(const SubmapID submap_id) const;
 
   // Creates a new submap on the top of the collection
-  void createNewSubMap(const Transformation &T_M_S, SubmapID submap_id);
+  void createNewSubMap(const Transformation &T_M_S, const SubmapID submap_id);
   void createNewSubMap(const Transformation &T_M_S);
 
   // Create a new submap which duplicates an existing source submap
-  bool duplicateSubMap(const cblox::SubmapID source_submap_id, const cblox::SubmapID new_submap_id);
-
-  // Gets a const reference to a raw submap
-  const TsdfSubmap &getSubMap(size_t sub_map_index) const {
-    CHECK_LT(sub_map_index, tsdf_sub_maps_.size());
-    return *(tsdf_sub_maps_[sub_map_index]);
-  }
-
-  // Gets a const reference to the raw submap vector
-  const std::vector<TsdfSubmap::Ptr> &getSubMaps() const {
-    return tsdf_sub_maps_;
-  }
-
-  // Flattens the collection map down to a normal TSDF map
-  TsdfMap::Ptr getProjectedMap() const;
-
-  // Gets the pose of the patch on the tip of the collection
-  const Transformation &getActiveSubMapPose() const {
-    return tsdf_sub_maps_.back()->getPose();
-  }
-
-  // Gets the ID of the patch on the tip of the collection
-  const SubmapID getActiveSubMapID() const {
-    return tsdf_sub_maps_.back()->getID();
-  }
-
-  // Gets a pointer to the active tsdf_map
-  TsdfMap::Ptr getActiveTsdfMapPtr() {
-    return tsdf_sub_maps_.back()->getTsdfMapPtr();
-  }
-  // Gets a reference to the active tsdf_map
-  const TsdfMap &getActiveTsdfMap() const {
-    return tsdf_sub_maps_.back()->getTsdfMap();
-  }
-
-  // Gets a reference to the active tsdf_map
-  const TsdfSubmap &getActiveTsdfSubMap() const {
-    return *(tsdf_sub_maps_.back());
-  }
-
-  // Associates a to the active submap
-  void associateIDToActiveSubmap(const SubmapID submap_id) {
-    id_to_submap_[submap_id] = tsdf_sub_maps_.back();
-  }
-
-  // Gets the tsdf submap associated with the passed ID
-  bool getAssociatedTsdfSubMapID(const SubmapID submap_id,
-                                 SubmapID *submap_id_ptr) const;
+  bool duplicateSubMap(const SubmapID source_submap_id,
+                       const SubmapID new_submap_id);
+
+  // Gets a const pointer to a raw submap
+  // NOTE(alexmillane): This function hard fails when the submap doesn't
+  // exist... This puts the onus on the caller to call exists() first. I don't
+  // like this but I can't see a solution.
+  const TsdfSubmap &getSubMap(const SubmapID submap_id) const;
+  // Note(alexmillane): Unlike the above this function returns a nullptr when
+  // the map doesn't exist. No hard crash.
   TsdfSubmap::ConstPtr getTsdfSubmapConstPtrById(
       const SubmapID submap_id) const;
+  // A list of the submaps
+  const std::vector<TsdfSubmap::Ptr> getSubMaps() const;
```

Comment:
Since you can get a vector of Ids, I'm not really sure why to allow this? I guess if you want to quicker merge the map together without doubling the look-up time...

Reply:
Hmm. Agreed. I should really remove this, but because the mesher makes frequent use of it, and I'm lazy, I won't (for now).